plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/ui/navigator/model/AppEngineResourceElement.java

```diff
@@ -23,17 +23,18 @@
 import org.eclipse.jface.viewers.StyledString;
 
 /**
- * A representation for App Engine-specific configuration files. Amongst other things, it mplements
- * IAdaptable to expose the configuration file to enable Eclipse's <em>Open</em> functionality.
+ * A model representation of specific App Engine configuration files. Amongst other things, it
+ * implements IAdaptable to expose the configuration file to enable Eclipse's <em>Open</em>
+ * functionality.
  */
 public abstract class AppEngineResourceElement implements IAdaptable {
   private final IProject project;
   private final IFile file;
 
-  public AppEngineResourceElement(IProject project, IFile file) {
-    this.project = Preconditions.checkNotNull(project);
+  public AppEngineResourceElement(IFile file) {
     this.file = Preconditions.checkNotNull(file);
     Preconditions.checkState(file.exists());
+    this.project = file.getProject();
```

Comment:
How about removing the field and make `getProject()` return `file.getProject()`? The field doesn't seem so useful.

Reply:
ok